---
title: "Differential gene expression analysis of adjacent melanoma tissue and melanoma cell lines."
output: html_notebook
---

load in libraries
```{r}

library(data.table)
library(tidyverse)
library(readr)
#library(textshape)
library(pheatmap)
library(RColorBrewer)
library(DESeq2)
library(ggplot2)
library(dplyr)
library(tidyr)

library(EnhancedVolcano)
library(sva)


#library(ggrepel)
library(textshaping)

library(UpSetR)
```


TISSUE SAMPLES

Read in the count data of the tissue samples
```{r}
# set working directory

setwd("C:/Users/Amara/Documents/cell_lines")

# read in count data
countdata <- read.table("C:/Users/Amara/Documents/cell_lines/tissue_counts_data.txt", header =TRUE)

countdata
```

```{r}
countdata
```


Rename the Geneid column

```{r}
colnames(countdata)[1] ="gene_id"

countdata
```



```{r}
dim(countdata)
```
The dataset has 62703 rowss and 18 columns..


Read in the gene name files
```{r}
setwd("C:/Users/Amara/Documents/cell_lines")
genes_name <- read.table("C:/Users/Amara/Documents/cell_lines/genomic_feature.txt", header =TRUE)

genes_name
```
Merge the count data and the gene name files.

```{r}
genes_name
```




```{r}

count_genename_merged <- inner_join(countdata, genes_name, by = c("gene_id"))  # Applying inner_join() function

count_genename_merged  
```

combine the geneid and the gene name column

```{r}
count_genename_combine <- within(count_genename_merged, combined <- paste(gene_id, gene_name, sep='&'))
count_genename_combine
```

```{r}
# rearrange 
# move the  combined column to column 1

countData <- count_genename_combine %>% relocate(combined, .before = gene_id)

countData

```


Data cleaning

```{r}



# remove the gene_id,gene_name and gene type columns.
countData <-countData[-c(2,20,21)]

countData

```


Remove the name extensions
```{r}
#remove the prefix
colnames(countData)<- gsub("X.home.stud8.NAS.mapping.gencode_ref.star_mapped_files.","",colnames(countData))

# remove the suffix
colnames(countData)<- gsub(".Aligned.sortedByCoord.out.bam","",colnames(countData))

countData
```

Convert the combined column to rownames
```{r}
# convert the combined column to rownames
countData <- column_to_rownames(countData,"combined")

countData
```

Change the sample names to describe the data better
Creating new sample names(column names),
```{r}
columns <- c("C_retained", "A_lost", "A_retained", "J_lost", "J_retained","I_lost_40","I_lost_41","I_retained","H_lost","H_retained","F_lost","F_retained","E_lost", "E_retained","D_lost","D_retained","C_lost" )

columns
```


```{r}
#columns <- c("C", "A.", "A", "J.", "J","I.","I..","I","H.","H","F.","F","E.", "E","D.","D","C." )

#columns
```



Insert the new sample names
```{r}

colnames(countData)[c(1:17)] = columns
countData
```
```{r}
View(countData)
```

Move the C_lost column to the first column to align with others.

```{r}

# move C_lost before the C_retained.
countData <- countData %>% relocate(C_lost, .before = C_retained)

countData

```


```{r}

# move C_lost before the C_retained.
#countData <- countData %>% relocate(C., .before = C)

#countData

```




Creating colData
```{r}
# creating conditions

condition = factor(c("lost","retained","lost","retained","lost","retained","lost","lost","retained","lost","retained","lost","retained","lost","retained","lost","retained"))
condition
```



```{r}
# create groups
group<- factor(c("C","C","A","A","J","J","I","I","I","H","H","F","F","E","E","D","D"))

group
```

Create coldata

```{r}

# create coldata using the group and condition
colData = data.frame(row.names=colnames(countData),condition,group)
colData
```

write to file
```{r}
write.table(colData, file="tissue_coldata.txt", sep="\t", quote=F, col.names=NA)
```

```{r}
getwd()
```


Construct a deseq2dataset object

Here, the design is what you want to compare,eg lost and retained. Use the condition as design.
```{r}
dds = DESeqDataSetFromMatrix(countData = countData,
                             colData = colData,
                             design = ~ condition)
dds
```
```{r}
dim(dds)
```


Prefiltering the dataset

Remove rows of the DESeqDataSet that have low counts
```{r}
# keep rows with rowsum greater than 5
dds <- dds[rowSums(counts(dds)) > 5,]
dim(dds)
```

Quality control,Exploratory analysis and Visualization

Quality Control
The next step in the DESeq2 workflow is QC which checks to ensure that the samples look good.


rlog & vst transformation


```{r}

# rlog transformation

rld <- rlog(dds)
head(assay(rld))

```


To see the distribution of the values:

```{r}
hist(assay(rld))
```


PCA

It shows the relationship between the samples and how they separate from each other.
PCA plot of the rlog transformed data

```{r}
plotPCA(rld, intgroup = c("condition","group"))
plotPCA(rld, intgroup = c("condition"))
#plotPCA(rld, intgroup = c("group"))

# put the plot in a variable
p_condition <- plotPCA(rld, intgroup=c("condition"))
p_group <- plotPCA(rld, intgroup = c("condition","group"))

 # label the samples with their names
 p_condition + geom_text(aes(label = name))
 p_group + geom_text(aes(label = name)
  )
```




vst transformed data
```{r}

vsd <- vst(dds)

# recapture the last plot in a variable
x <- plotPCA(vsd, intgroup=c("condition"))
y <- plotPCA(vsd, intgroup = c("condition","group"))
z <- plotPCA(vsd, intgroup=c("group"))

# label the sample names
 x + geom_text(aes(label = name))
 y + geom_text(aes(label = name))
 z + geom_text(aes(label = name))
```
The data seemed to cluster by group

```{r}
pcaData <- plotPCA(vsd, intgroup=c("condition"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
ggplot(pcaData, aes(PC1, PC2, color=condition)) +
  geom_point(size=3) +
  geom_text_repel(aes(label = name), nudge_x = -1, nudge_y = 0.2, size = 7) +
  #geom_text(aes(label = name))+
  #theme(legend.title = element_text(size=14))+
  #theme(legend.text = element_text(size=13))+
  theme(axis.title = element_text(size = 20))+
  theme(text = element_text(size = 20))+
  theme(axis.text = element_text(size = 20))+
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed()
```




BATCH EFFECT
correct for batch effect using Limma

batch = group


specify ~ condition + group meaning to test for the effect of the condition controlling for the
effect of different patients.


Create dds with condition and group
```{r}
dds_limma <- DESeqDataSetFromMatrix(countData = countData,colData = colData,design = ~ condition + group)

```


```{r}
dds_limma
```

Here, remove rows of the DESeqDataSet that have low counts
```{r}
# keep rows with rowsum greater than 5
dds_limma <-dds_limma[rowSums(counts(dds_limma)) > 5,]
dim(dds_limma)
```


PCA plot showing rlog transformation of the removed batch effect data

rlog before removing batch effect

```{r}
rld_limma <- rlog(dds_limma)
rld_limma$group

```

```{r}
plotPCA(rld_limma, intgroup = c("condition","group"))
plotPCA(rld_limma, intgroup = c("condition"))
plotPCA(rld_limma, intgroup = c("group"))

# put the plot in a variable
l_condition <- plotPCA(rld_limma, intgroup=c("condition"))
l_group <- plotPCA(rld_limma, intgroup = c("condition","group"))

 # label the samples with their names
 l_condition + geom_text(aes(label = name))
 l_group + geom_text(aes(label = name))
```

rlog after removing batch effect
```{r}
assay(rld_limma) <- limma::removeBatchEffect(assay(rld_limma), rld_limma$group)
head(rld_limma)
l_condition <- plotPCA(rld_limma, "condition")
l_condition + geom_text(aes(label = name))



```

```{r}
pcaData <- plotPCA(rld_limma, intgroup=c("condition"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
ggplot(pcaData, aes(PC1, PC2, color=condition)) +
  geom_point(size=3) +
  geom_text_repel(aes(label = name), nudge_x = -1, nudge_y = 0.2, size = 6) +
  #geom_text(aes(label = name))+
  #theme(legend.title = element_text(size=14))+
  #theme(legend.text = element_text(size=13))+
  theme(axis.title = element_text(size = 20))+
  theme(text = element_text(size = 20))+
  theme(axis.text = element_text(size = 20))+
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed()
```


VST

vst before removing batch effect
```{r}
vsd_limma <- vst(dds_limma,blind=FALSE) 



plotPCA(vsd_limma, intgroup = c("condition"))

# put the plot in a variable
l_condition <- plotPCA(vsd_limma, intgroup=c("condition"))
l_group <- plotPCA(vsd_limma, intgroup = c("condition","group"))

 # label the samples with their names
 l_condition + geom_text(aes(label = name))
 l_group + geom_text(aes(label = name))

```


vst after removing batch effect
```{r}
assay(vsd_limma) <- limma::removeBatchEffect(assay(vsd_limma), vsd_limma$group)
head(vsd_limma)
v_condition <- plotPCA(vsd_limma, "condition") 
v_condition + geom_text(aes(label = name))

plotPCA(vsd_limma, "group")

```

```{r}
pcaData <- plotPCA(vsd_limma, intgroup=c("condition"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
ggplot(pcaData, aes(PC1, PC2, color=condition)) +
  geom_point(size=3) +
  geom_text_repel(aes(label = name), nudge_x = -1, nudge_y = 0.2, size = 7) +
  #geom_text(aes(label = name))+
  #theme(legend.title = element_text(size=14))+
  #theme(legend.text = element_text(size=13))+
  theme(axis.title = element_text(size = 20))+
  theme(text = element_text(size = 20))+
  theme(axis.text = element_text(size = 20))+
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed()
```




Hierarchical Clustering Heatmap
Similar to PCA, hierarchical clustering is another method for identifying patterns in a dataset and potential outliers. The heatmap displays the correlation of gene expression for all pairwise combinations of samples in the dataset. 

Heatmap assess overall similarity between samples.


And now to plot the distance values as a heatmap using pheatmap function.

Before batch effect
```{r}

sampleDists <- dist(t(assay(rld)))

sampleDists

sampleDistsMatrix <- as.matrix(sampleDists)

colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)

pheatmap(sampleDistsMatrix, clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors,fontsize = 15)

```

After removing batch effect

```{r}
sampleDists <- dist(t(assay(rld_limma)))

sampleDists


sampleDistsMatrix <- as.matrix(sampleDists)

colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)

pheatmap(sampleDistsMatrix, clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors,fontsize = 15)
```
The color represents the distance between the sampples, the C samples shows more distance across all samples.

After the exploration of the DeseqDataObject;

Run the Deseq function
```{r}
dds <- DESeq(dds)
```

Export normalized read counts,Deseq normalized.

```{r}
tissue_normalized_counts <- counts(dds,normalized=TRUE)
write.table(tissue_normalized_counts,"tissue_normalized.lost.retained.txt")

```



Save the result of the deseq in res
```{r}
res <- results(dds,contrast=c("condition","lost","retained"))
res

```

Remove NA
```{r}
#drop nas first
dim(res)

res <- na.omit(res)
res

dim(res)
```



Export the result

```{r}

write.table(res, file="main_res_tissue_sample1.txt")
```



Explore results

At a FDR of 0.1, we have 3 genes up-regulated and 18 genes down-regulated in the lost vs retained.
```{r}
summary(res)
```

```{r}
# check the number of significantly regulated genes in res
sum(res$padj < 0.1, na.rm=TRUE)
```
  
Check the number of up-regulated and down-regulated
```{r}
table(res$padj < 0.1 & res$log2FoldChange > 1)   # significantly upregulated

table(res$padj < 0.1 & res$log2FoldChange < 1)   # significantly downregulated
```

FILTER

Filter the significantly expressed genes
Genes with padj value less than 0.1
```{r}

signgenes<- res[res$padj < 0.1,]
dim(signgenes)

signgenes
```



```{r}

write.table(signgenes, file="tissue_signgenes.txt")
```




Up-regulated genes
Orders by the most up-regulated genes using l2fC
```{r}
 head(signgenes[order(signgenes$log2FoldChange,decreasing = T),])
```




Significant genes ordered by decreasing log2FoldChange

The genes with positive log2foldchange are up-regulated and those with negative log2foldchange are down-regulated.
```{r}
Signgenes <- (signgenes[order(signgenes$log2FoldChange,decreasing = T),])

Signgenes <- as.data.frame(Signgenes)
view(Signgenes)
```


```{r}
Signgenes
```



PLOTTING RESULTS


plots MA
```{r}
plotMA(res,cex.lab=1.5, cex.axis=1.5, cex.main=1.5, cex.sub=1.5)



plotMA(res, cex.lab=2.0,cex.axis=1.5,cex=0.8,colNonSig = "gray60",
  colSig = "red",
  colLine = "grey40",)
```
Shows the genes that are differentially expressed., the blue points are significantly expressed. The blue points above the line are up-regulated and the points below the line are down_regulated. 


Dispersion estimate plot
```{r}
#plot dispersion
plotDispEsts(dds)
```
Higher counts have low dispersion,low count have high disparity


Volcano Plot


```{r}
dim(res)

```


Convert to data frame
```{r}
de_genes <- as.data.frame(res)

de_genes
```



```{r}
# Create a new column for the diff expressed genes and label it "NO"
de_genes$diffexpressed <- "NO"

# labels the significantly up_regulated genes as UP and the significantly downregulated genes as DOWN. The rest of the column as NO.
de_genes$diffexpressed[de_genes$log2FoldChange > 1 & de_genes$padj < 0.1] <- "UP"
de_genes$diffexpressed[de_genes$log2FoldChange < 1 & de_genes$padj < 0.1] <- "DOWN"

```
 
```{r}

de_genes[de_genes$diffexpressed  == 'UP',]
```
```{r}
de_genes[de_genes$diffexpressed  == 'DOWN',]
```




```{r}
de_genes
de_genes$delabel <- NA #label
```



Plot volcano plot with ggplot

```{r}
ggplot(data=de_genes,aes(x=log2FoldChange,y =-log10(pvalue),col=diffexpressed,label=delabel))+
  geom_point() +
  theme_minimal()+
  geom_text_repel()+
  scale_color_manual(values=c("blue","black","red"))+
  theme(text=element_text(size=20))
```

Enhanced Volcano plots

Split the combined column to get the gene name
```{r}
# convert res to a dataframe

res_vol <- as.data.frame(res)

# convert rownames to a column combined
res_vol <- setDT(res_vol, keep.rownames = "combined")

#split the combined column
res_vol <- separate(res_vol, combined , c("ENSG", "NAME"), sep="&", remove=TRUE, convert=FALSE, extra="warn", fill="warn")

```

```{r}
res_vol
```



Plot
```{r}
EnhancedVolcano(res_vol, 
    x="log2FoldChange",
    y="padj",
    lab=NA,
    title= "Lost versus Retained",
    pCutoff = 0.1,
    FCcutoff = 1,
    pointSize = 3.0,
    labSize = 6.0,
    
    axisLabSize  = 20,
    legendLabSize = 15,
    legendIconSize = 5.0)

```




Heatmap of the DEGs




```{r}

#selects the significant genes
sig_genes <- rownames(res[res$padj<0.1 & !is.na(res$padj), ])

sig_genes_mat <- assay(rld)[sig_genes,]
  
  
df <- as.data.frame(colData(rld)[,c("condition","group")])
pheatmap(sig_genes_mat, annotation_col=df,fontsize=9,fontsize_number = 0.4 * fontsize)

```
```{r}
signgeness <- as.data.frame(signgenes)

signgeness <- signgeness[order(signgeness$log2FoldChange),]

View(signgeness)
```




Extract the significant genes
```{r}
sign_lost_retained <- res[res$padj < 0.1,]
sign_lost_retained

dim(sign_lost_retained)
```

Convert to dataframe

```{r}
sign_lost_retained_df <- as.data.frame(sign_lost_retained)


# convert the rows to column name combined

sign_lost_retained <- setDT(sign_lost_retained_df, keep.rownames = "geneid")



```

```{r}
sign_lost_retained
```


Upregulated
```{r}

# significantly up-regulated genes

up_sign_lost_retained <- sign_lost_retained[sign_lost_retained$log2FoldChange > 1,]

up_sign_lost_retained

dim(up_sign_lost_retained)
```

downregulated

```{r}

# significantly up-regulated genes

down_sign_lost_retained <- sign_lost_retained[sign_lost_retained$log2FoldChange < 1,]

down_sign_lost_retained

dim(down_sign_lost_retained)
```





ANALYSIS OF THE MELANOMA CELL LINES.

Read in the melanoma cell lines

```{r}
#setwd("C:/Users/Amara/Documents/cell_lines")

mela_model <- read_delim("C:/Users/Amara/Documents/cell_lines/all_melanoma_featureCountsTable_edited.txt", "\t", escape_double = FALSE, trim_ws = TRUE) 


```

```{r}
View(mela_model)
```


```{r}
# view column names
colnames(mela_model)
```
Remove columns that are not useful for the analysis.

Keeping the combined column, and the 6 cell line samples for the analysis.
```{r}
mela_model

# remove from 2nd column till the 10th column.
mela_model<- mela_model[,-2:-10]

# remove the 14th till the 21st column ,which contained samples not needed for the analysis.
mela_model<- mela_model[,-14:-21]
mela_model
```


```{r}
dim(mela_model)
```
The remaining dataframe contains 60725 genes and 12 samples.

Convert column name to row names
```{r}
# convert the combined column to rownames
mela_model <- column_to_rownames(mela_model,"combined")
mela_model
```


Remove the name extensions
```{r}
colnames(mela_model)<- gsub("/Aligned.sortedByCoord.out.bam","",colnames(mela_model))

mela_model
```

```{r}
View(mela_model)
```



Create coldata for the melanoma cell lines.


```{r}
 
# creating columns of the coldata
  

  
BRAF <- factor(c("WT","WT","WT","WT","V600E","V600E","WT","WT","V600E","V600E","V600E","V600E"))
  

PTEN_Mutation <- factor(c("WT","WT","WT","WT","Hem_deletion","Hem_deletion","WT","WT","Hem_deletion","Hem_deletion","Hom_deletion","Hom_deletion"))

NRAS <- factor(c("Q61K","Q61K","WT","WT","WT","WT","Q61L","Q61L","WT","WT","WT","WT"))

P53 <- factor(c("WT","WT","Mut","Mut","WT","WT","WT","WT","WT","WT","WT","WT"))

CDK4 <- factor(c("WT","WT","WT","WT","K22Q","K22Q","WT","WT","WT","WT","WT","WT"))

```





```{r}
# create a dataframe using the columns of the coldata

melanoma_coldata = data.frame(row.names=colnames(mela_model),BRAF,PTEN_Mutation,NRAS,P53,CDK4)
melanoma_coldata 

```

```{r}
View(melanoma_coldata)
```

write to file
```{r}
write.table(melanoma_coldata, file="melanoma_coldata.txt", sep="\t", quote=F, col.names=NA)
```


Check that the row names of the melanoma_coldata equals the column names of the count table (mela_model)

```{r}
all(colnames(mela_model$counts) == rownames(melanoma_coldata))
```



Construct a deseq2dataset object(dds) for the cell lines.

```{r}

# a dds object using the PTEN_Mutation as design.

dds_mela = DESeqDataSetFromMatrix(countData=mela_model,
                             colData=melanoma_coldata,
                             design = ~PTEN_Mutation)
dds_mela

```
keep rows with row sum value greater than 5
```{r}
dim(dds_mela)

dds_mela <- dds_mela[rowSums(counts(dds_mela)) > 5,]
dim(dds_mela)
```

Quality Control

For visualization, there is need to moderate the variance across the mean by applying the rlog transformation to the dds object. This is only used for visualization eg. PCA and heirarchical clustering(heatmap).
```{r}
# Transform counts for data visualization

rld_mela <- rlog(dds_mela,blind=FALSE)

```


Principal components analysis (PCA)

Using the PCA to visualize the different PTEN mutation.

```{r}
# plot PCA for PTEN_Mutation

m_condition <- plotPCA(rld_mela, intgroup = c("PTEN_Mutation"))

# gives title to the plot
m_condition + labs(title = paste("PCA Plot of the Cell Lines for PTEN_Mutation")) 

#label points with the sample name
m_condition + geom_text(aes(label = name)) 




```
The samples cluster by replicates. In the PTEN_Mutation plot, the 2 WT that has almost same mutations cluster more than other WT with another mutation on NRAS.


```{r}
pcaData_mela <- plotPCA(rld_mela, intgroup=c("PTEN_Mutation"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData_mela, "percentVar"))
ggplot(pcaData_mela, aes(PC1, PC2, color=PTEN_Mutation)) +
  geom_point(size=3) +
  geom_text_repel(aes(label = name), nudge_x = -1, nudge_y = 0.2, size = 8) +
  #geom_text(aes(label = name))+
  #theme(legend.title = element_text(size=14))+
  #theme(legend.text = element_text(size=13))+
  theme(text = element_text(size = 27))+
  theme(axis.text = element_text(size = 27))+
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed()
```


rlog after removing batch effect
```{r}
assay(rld_mela) <- limma::removeBatchEffect(assay(rld_mela), rld_mela$group)
head(rld_mela)
m_condition <- plotPCA(rld_mela, "PTEN_Mutation")
m_condition + geom_text(aes(label = name))



```
There was no batch effect




Heatmap

using correlation
```{r}
# Extract the rlog matrix from the object

rld_mat <- assay(rld_mela)

# Compute pairwise correlation values
rld_corr <- cor(rld_mat)   
head(rld_corr)


### Plot heatmap
pheatmap(rld_corr)
```
Using sample distance
```{r}
sampleDists <- dist(t(assay(rld_mela)))

sampleDists

sampleDistsMatrix <- as.matrix(sampleDists)

colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)

pheatmap(sampleDistsMatrix, clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors,fontsize = 15)
```


Also, similar to the PCA plot, the samples cluster by sample group
Together, these plots suggest that the data are of good quality.

```{r}
melanoma_coldata

```





Run analysis(DESeq function)
```{r}
dds_mela <- DESeq(dds_mela)

```

Export the normalized count

```{r}

cell_line_normalized_counts <- counts(dds_mela, normalized=TRUE)


write.table(cell_line_normalized_counts, file="cell-lines_normalized_counts.txt", sep="\t", quote=F, col.names=NA)

```


Convert the normalized counts to a dataframe.
```{r}
cell_line_normalized_counts_df <- as.data.frame(cell_line_normalized_counts)

cell_line_normalized_counts_df
```

 
```{r}
# convert the rownames to column name combined 
cell_line_normalized_counts_df <- setDT(cell_line_normalized_counts_df, keep.rownames = "combined")

cell_line_normalized_counts_df
```



Save the result of the deseq in res

For the different comparison of the PTEM mutation, create contrasts to perform testing between different conditions.

1
###################################################
###################################################

Comparison of the Hem_deletions and WTs.

```{r}
res_Hem_WT <- results(dds_mela,contrast = c("PTEN_Mutation","Hem_deletion","WT"))

res_Hem_WT
```



Remove NA
```{r}

dim(res_Hem_WT)

res_Hem_WT <- na.omit(res_Hem_WT)
res_Hem_WT

dim(res_Hem_WT)

```
After removing NAs, the remaining result has 20901 genes.



Export the result

```{r}

write.table(res_Hem_WT, file="new_res_Hem_WT.txt")
```


```{r}
summary(res_Hem_WT)
```



Explore result

MA plot plots the log2 fold changes against the mean of normalized counts for all the samples. 


The blue points shows the differentially expressed genes with adjusted pvalue less than 0.01. The blue points above zero are the significantly up-regulated genes and the ones below zero are the significantly down-regulated genes. Points that fall out of the window are represented as triangles pointing either up or down showing the direction of their fold change.


```{r}
plotMA(res_Hem_WT)

plotMA(res_Hem_WT, ylim= c(-8,8),cex.lab=2.3,cex.axis=2.3)
```


Enhanced Volcano plots

Split the combined column to get the gene name
```{r}
# convert res_Hem_WT to a dataframe

res_HemWT <- as.data.frame(res_Hem_WT)

# convert rownames to a column combined
res_HemWT <- setDT(res_HemWT, keep.rownames = "combined")

#split the combined column
res_HemWT <- separate(res_HemWT, combined , c("ENSG", "NAME"), sep="&", remove=TRUE, convert=FALSE, extra="warn", fill="warn")

```


Plot
```{r}
EnhancedVolcano(res_HemWT, x="log2FoldChange",y="padj",lab=NA, title= "Hem versus WT")

```

```{r}
EnhancedVolcano(res_HemWT, 
    x = 'log2FoldChange', FCcutoff=1, 
    y = 'padj',
     lab=res_HemWT$NAME,
    title= "Hem versus WT",
    pCutoff = 0.1,
   # FCcutoff = 0,
    pointSize = 3.0,
    axisLabSize  = 40,
   
    legendLabSize = 30,
    legendIconSize = 10.0,
    
    labSize = 11.0)
```



Significant results
Extracting results with padj of 0.01

2305 genes have padj value of 0.01. These are the significant genes in the comparison of Hemizygous deletion and wildtype of the PTEN mutation.

```{r}
sign_Hem_WT <- res_Hem_WT[res_Hem_WT$padj < 0.01,]
sign_Hem_WT

dim(sign_Hem_WT)
```
Convert to dataframe

```{r}
sign_Hem_WT_df <- as.data.frame(sign_Hem_WT)


# convert the rows to column name combined

sign_Hem_WT_df <- setDT(sign_Hem_WT_df, keep.rownames = "combined")

sign_Hem_WT_df

```


Extract the up and down regulated genes from the significant genes.

```{r}

# significantly up-regulated genes

up_sign_Hem_WT <- sign_Hem_WT_df[sign_Hem_WT_df$log2FoldChange > 1,]

up_sign_Hem_WT

dim(up_sign_Hem_WT)
```

```{r}
# significantly down-regulated genes

down_sign_Hem_WT <- sign_Hem_WT_df[sign_Hem_WT_df$log2FoldChange < 1,]


down_sign_Hem_WT

dim(down_sign_Hem_WT)

```




2

#############################################################
#############################################################



Comparison of the Hom_deletion and WT.

```{r}
res_Hom_WT <- results(dds_mela,contrast = c("PTEN_Mutation","Hom_deletion","WT"))

res_Hom_WT
```

Remove NA
```{r}

dim(res_Hom_WT)

res_Hom_WT <- na.omit(res_Hom_WT)
res_Hom_WT

dim(res_Hom_WT)

```
```{r}
res_Hom_WT
```



Export the result



```{r}
write.table(res_Hom_WT, file="new_res_Hom_WT.txt")
```



```{r}
summary(res_Hom_WT)
```

Explore result


MA plot plots the log2 fold changes against the mean of normalized counts for all the samples. 

MA Plot Homozygous deletion and Wildtype for PTEN_Mutation

The blue points shows the differentially expressed genes with adjusted pvalue less than 0.01. The blue points above zero are the significantly up-regulated genes and the ones below zero are the significantly down-regulated genes. Points that fall out of the window are represented as triangles pointing either up or down showing the direction of their fold change.
```{r}

plotMA(res_Hom_WT)

plotMA(res_Hom_WT, ylim= c(-8,8),cex.lab=2.3,cex.axis=2.3)
```

Volcano plots



Enhanced Volcano plots

Split the combined column to get the gene name
```{r}
# convert res_Hom_WT to a dataframe

res_HomWT <- as.data.frame(res_Hom_WT)

# convert rownames to a column combined
res_HomWT <- setDT(res_HomWT, keep.rownames = "combined")

#split the combined column
res_HomWT <- separate(res_HomWT, combined , c("ENSG", "NAME"), sep="&", remove=TRUE, convert=FALSE, extra="warn", fill="warn")

```


Plot

```{r}
EnhancedVolcano(res_HomWT, x="log2FoldChange",y="padj",lab=NA, title= "Hom versus WT")

```



Significant results
Extracting results with padj of 0.01


```{r}
sign_Hom_WT <- res_Hom_WT[res_Hom_WT$padj < 0.01,]
sign_Hom_WT

dim(sign_Hom_WT)
```
Convert to dataframe 

```{r}
sign_Hom_WT_df <- as.data.frame(sign_Hom_WT)


# convert the rows to column name combined

sign_Hom_WT_df <- setDT(sign_Hom_WT_df, keep.rownames = "combined")



```

```{r}
sign_Hom_WT_df
```


Extract the up and down regulated genes from the significant genes.

```{r}

# significantly up-regulated genes

up_sign_Hom_WT <- sign_Hom_WT_df[sign_Hom_WT_df$log2FoldChange > 1,]

up_sign_Hom_WT

dim(up_sign_Hom_WT)
```

```{r}
# significantly down-regulated genes

down_sign_Hom_WT <- sign_Hom_WT_df[sign_Hom_WT_df$log2FoldChange < 1,]

down_sign_Hom_WT

dim(down_sign_Hom_WT)

```



##################################################################


Create Coldata for the melanoma models.
To compare the two Hem_deletion in the PTEN_Mutation column, add the specific mutation (Mut_W274X and Mut_V343E)  while creating the coldata in order to differentiate them.
```{r}
 
  
#cell_line <- factor(c("Sbcl2","WM3211","WM793","WM1366","WM1158","WM9"))

  
BRAF <- factor(c("WT","WT","WT","WT","V600E","V600E","WT","WT","V600E","V600E","V600E","V600E"))
  

PTEN_Mutation_Hem <- factor(c("WT","WT","WT","WT","Hem_deletion_Mut_W274X","Hem_deletion_Mut_W274X","WT","WT","Hem_deletion_Mut_V343E","Hem_deletion_Mut_V343E","Hom_deletion","Hom_deletion"))

NRAS <- factor(c("Q61K","Q61K","WT","WT","WT","WT","Q61L","Q61L","WT","WT","WT","WT"))

P53 <- factor(c("WT","WT","Mut","Mut","WT","WT","WT","WT","WT","WT","WT","WT"))

CDK4 <- factor(c("WT","WT","WT","WT","K22Q","K22Q","WT","WT","WT","WT","WT","WT"))

```




```{r}
# create coldata

melanoma_coldata_Hem = data.frame(row.names=colnames(mela_model),BRAF,PTEN_Mutation_Hem,NRAS,P53,CDK4)
melanoma_coldata_Hem

```


```{r}
write.table(melanoma_coldata_Hem, file="melanoma_coldata_Hem.txt")
```


Construct a deseq2dataset object

```{r}
dds_PTEN_Hem = DESeqDataSetFromMatrix(countData=mela_model,
                             colData=melanoma_coldata_Hem,
                             design= ~PTEN_Mutation_Hem)
dds_PTEN_Hem

```

Run DESeq

```{r}
dds_PTEN_Hem <- DESeq(dds_PTEN_Hem)

```



Results

4
###########################################
###########################################

Comparing Hem_deletion_W274X and WT

```{r}

res_Hem_W274X_WT <- results(dds_PTEN_Hem,contrast = c("PTEN_Mutation_Hem","Hem_deletion_Mut_W274X","WT"))
res_Hem_W274X_WT
```


Drop NA
```{r}

dim(res_Hem_W274X_WT)

res_Hem_W274X_WT <- na.omit(res_Hem_W274X_WT)
res_Hem_W274X_WT

dim(res_Hem_W274X_WT)
```


Export the result


```{r}

write.table(res_Hem_W274X_WT, file="res_Hem_W274X_WT.txt")
```



```{r}
summary(res_Hem_W274X_WT)
```


plots MA
```{r}


plotMA(res_Hem_W274X_WT)

plotMA(res_Hem_W274X_WT, ylim= c(-8,8),cex.lab=2.3,cex.axis=2.3)
```

Enhanced Volcano plots

Split the combined column to get the gene name
```{r}
# convert res_Hem_W274X_WT to a dataframe

res_Hem_W274XWT <- as.data.frame(res_Hem_W274X_WT)

# convert rownames to a column combined
res_Hem_W274XWT <- setDT(res_Hem_W274XWT, keep.rownames = "combined")

#split the combined column
res_Hem_W274XWT <- separate(res_Hem_W274XWT, combined , c("ENSG", "NAME"), sep="&", remove=TRUE, convert=FALSE, extra="warn", fill="warn")

```
Plot
```{r}
EnhancedVolcano(res_Hem_W274XWT, x="log2FoldChange",y="padj",lab=NA, title= "HemW274X versus WT")

```





Significant results

Extracting results with padj of 0.01




```{r}
sign_Hem_W274X_WT <- res_Hem_W274X_WT[res_Hem_W274X_WT$padj < 0.01,]
sign_Hem_W274X_WT

dim(sign_Hem_W274X_WT)
```

Convert to dataframe

```{r}
sign_Hem_W274X_WT_df <- as.data.frame(sign_Hem_W274X_WT)


# convert the rows to column name combined

sign_Hem_W274X_WT_df <- setDT(sign_Hem_W274X_WT_df, keep.rownames = "combined")



```

```{r}
sign_Hem_W274X_WT_df
```



Extract the up and down regulated genes from the significant genes.

```{r}

# significantly up-regulated genes

up_sign_Hem_W274X_WT <- sign_Hem_W274X_WT_df[sign_Hem_W274X_WT_df$log2FoldChange > 1,]

up_sign_Hem_W274X_WT

dim(up_sign_Hem_W274X_WT)
```

```{r}
# significantly down-regulated genes

down_sign_Hem_W274X_WT <- sign_Hem_W274X_WT_df[sign_Hem_W274X_WT_df$log2FoldChange < 1,]

down_sign_Hem_W274X_WT

dim(down_sign_Hem_W274X_WT)
```







5

###############################################
###############################################

Comparing Hem_deletion_V343E and WT

```{r}

res_Hem_V343E_WT <- results(dds_PTEN_Hem,contrast = c("PTEN_Mutation_Hem","Hem_deletion_Mut_V343E","WT"))
res_Hem_V343E_WT
```


Drop NA
```{r}

dim(res_Hem_V343E_WT) 

res_Hem_V343E_WT <- na.omit(res_Hem_V343E_WT)
res_Hem_V343E_WT

dim(res_Hem_V343E_WT)
```


Export the result


```{r}

write.table(res_Hem_V343E_WT, file="res_Hem_V343E_WT.txt")
```



```{r}
summary(res_Hem_V343E_WT)
```

plots MA
```{r}


plotMA(res_Hem_V343E_WT)

plotMA(res_Hem_V343E_WT, ylim= c(-8,8),cex.lab=2.3,cex.axis=2.3)
```


Enhanced Volcano plots

Split the combined column to get the gene name
```{r}
# convert res_Hem_V343E_WT to a dataframe

res_Hem_V343EWT <- as.data.frame(res_Hem_V343E_WT)

# convert rownames to a column combined
res_Hem_V343EWT <- setDT(res_Hem_V343EWT, keep.rownames = "combined")

#split the combined column
res_Hem_V343EWT <- separate(res_Hem_V343EWT, combined , c("ENSG", "NAME"), sep="&", remove=TRUE, convert=FALSE, extra="warn", fill="warn")

```

Plot
```{r}
EnhancedVolcano(res_Hem_V343EWT, x="log2FoldChange",y="padj",lab=NA, title= "Hem_V343E versus WT")

```




Significant results

Extracting significant genes with padj of 0.01


```{r}
sign_Hem_V343E_WT <- res_Hem_V343E_WT[res_Hem_V343E_WT$padj < 0.01,]
sign_Hem_V343E_WT

dim(sign_Hem_V343E_WT)
```



Convert to dataframe

```{r}
sign_Hem_V343E_WT_df <- as.data.frame(sign_Hem_V343E_WT)


# convert the rows to column name combined

sign_Hem_V343E_WT_df <- setDT(sign_Hem_V343E_WT_df, keep.rownames = "combined")



```

```{r}
sign_Hem_V343E_WT_df
```




Extract the up and down regulated genes from the significant genes.

```{r}

# significantly up-regulated genes

up_sign_Hem_V343E_WT <- sign_Hem_V343E_WT_df[sign_Hem_V343E_WT_df$log2FoldChange > 1,]

up_sign_Hem_V343E_WT

dim(up_sign_Hem_V343E_WT)
```

```{r}
# significantly down-regulated genes

down_sign_Hem_V343E_WT <- sign_Hem_V343E_WT_df[sign_Hem_V343E_WT_df$log2FoldChange < 1,]

down_sign_Hem_V343E_WT

dim(down_sign_Hem_V343E_WT)

```





##################################################
##################################################

The significant genes


```{r}
# For Hem_deletion and WT

sign_Hem_WT
```

```{r}
#For Hom_deletion and WT

sign_Hom_WT
```


```{r}

# For Hem_W274X and Hom_deletion

sign_Hem_W274X_WT
```

```{r}
# for  Hem_V343E and Hom_deletion

sign_Hem_V343E_WT

```




##############################################################
#####################################################

Upset Plot

As there are many comparisons,an upset plot will show common and unique DEGs for the different comparisons.

Extract the significantly up-regulated genes.
```{r}


# for the lost and retained result

lost_retained_up <- rownames(res)[res$log2FoldChange > 1 & res$padj < 0.1]


# cell lines

Hem_WT_up <- rownames(res_Hem_WT)[res_Hem_WT$log2FoldChange > 1 & res_Hem_WT$padj < 0.01]


Hom_WT_up <- rownames(res_Hom_WT)[res_Hom_WT$log2FoldChange > 1 & res_Hom_WT$padj < 0.01]



Hem_W274X_WT_up <- rownames(res_Hem_W274X_WT)[res_Hem_W274X_WT$log2FoldChange > 1 & res_Hem_W274X_WT$padj < 0.01]


Hem_V343E_WT_up <- rownames(res_Hem_V343E_WT)[res_Hem_V343E_WT$log2FoldChange > 1 & res_Hem_V343E_WT$padj < 0.01]




```

```{r}
lost_retained_up
```



Upset Plot

Creat a list of the comparisons to plot for the upregulated genes

```{r}
upsetList_up <- list(
  
  lostvsretained = sample(lost_retained_up),
  
  HemvsWT = sample(Hem_WT_up),
  HomvsWT = sample(Hom_WT_up),
  
  Hem_W274XvsWT = sample(Hem_W274X_WT_up),
  
  Hem_V343EvsWT = sample(Hem_V343E_WT_up)
  
)
```


```{r}
#upsetList_up
```


The plot for the 5 contrasts
```{r}
#upset plot for the upregulated genes

upset(fromList(upsetList_up),order.by = "freq",nsets=5)
```


#########################################################################################


Create a list of the significantly down-regulated genes.
```{r}

# for the lost and retained result

lost_retained_down <- rownames(res)[res$log2FoldChange < 1 & res$padj < 0.1]

# cell lines

Hem_WT_down <- rownames(res_Hem_WT)[res_Hem_WT$log2FoldChange < 1 & res_Hem_WT$padj < 0.01]


Hom_WT_down <- rownames(res_Hom_WT)[res_Hom_WT$log2FoldChange < 1 & res_Hom_WT$padj < 0.01]



Hem_W274X_WT_down <- rownames(res_Hem_W274X_WT)[res_Hem_W274X_WT$log2FoldChange < 1 & res_Hem_W274X_WT$padj < 0.01]


Hem_V343E_WT_down <- rownames(res_Hem_V343E_WT)[res_Hem_V343E_WT$log2FoldChange < 1 & res_Hem_V343E_WT$padj < 0.01]




```

```{r}
lost_retained_down
```



Create a list for the plot

Upset Plot

Create a list of the comparisons to plot for the downregulated genes

```{r}
upsetList_down <- list(
  
  lostvsretained = sample(lost_retained_down),
  
  HemvsWT = sample(Hem_WT_down),
  HomvsWT = sample(Hom_WT_down),
  
  
  Hem_W274XvsWT = sample(Hem_W274X_WT_down),
  
  Hem_V343EvsWT = sample(Hem_V343E_WT_down)
  
)
```


```{r}
#upsetList_down
```


The plot for the 5 contrasts
```{r}
#upset plot for the downregulated genes

upset(fromList(upsetList_down),order.by = "freq",nsets=5)
```












```{r}
sessionInfo()
```










